helpers/auth/admin_roles.py

```diff
@@ -1,81 +1,50 @@
+        location = Location.query.filter_by(id=kwargs['location_id']).first()
         if admin_details.location != location.name:
             raise GraphQLError("You are not authorized to make changes in " + location.name)  # noqa: E501
 
     def update_delete_rooms_create_resource(self, room_id):
         admin_details = get_user_from_db()
         location_query = location_join_room()
-        room_location = location_query.filter(RoomModel.id == room_id).first()  # noqa: E501
+        room_location = location_query.filter(
+            RoomModel.id == room_id).first()
         if admin_details.location != room_location.name:
             raise GraphQLError("You are not authorized to make changes in " + room_location.name)  # noqa: E501
 
-    def create_floor_update_delete_block(self, block_id):
-        admin_details = get_user_from_db()
-        location_query = location_join_block()
-        block_location = location_query.filter_by(id=block_id).first()
-        if admin_details.location != block_location.name:
-            raise GraphQLError("You are not authorized to make changes in " + block_location.name)  # noqa: E501
-
-    def update_delete_floor(self, floor_id):
-        admin_details = get_user_from_db()
-        location_query = location_join_floor()
-        floor_location = location_query.filter_by(id=floor_id).first()
-        if admin_details.location != floor_location.name:
-            raise GraphQLError("You are not authorized to make changes in " + floor_location.name)  # noqa: E501
-
-    def create_office(self, location_id):
-        admin_details = get_user_from_db()
-        location = Location.query.filter_by(id=location_id).first()
-        if location.name != admin_details.location:
-            raise GraphQLError("You are not authorized to make changes in " + location.name)  # noqa: E501
-
-    def check_office_location_create_wing(self, floor_id):
-        get_floor = FloorModel.query.filter_by(id=floor_id).first()
-        get_block = Block.query.filter_by(id=get_floor.block_id).first()  # noqa: E501
-        get_office = Office.query.filter_by(id=get_block.office_id).first()  # noqa: E501
-        get_location = Location.query.filter_by(id=get_office.location_id).first()  # noqa: E501
-        if get_location.name.lower() != 'lagos':
-            raise GraphQLError("This action is restricted to Lagos Office only")  # noqa: E501
-
-    def create_update_delete_wing(self):
-        admin_details = get_user_from_db()
-        if admin_details.location.lower() != 'lagos':
-            raise GraphQLError("This action is restricted to Lagos Office admin")  # noqa: E501
-
     def update_resource(self, resource_id, room_id):
         admin_details = get_user_from_db()
         location_query = location_join_resources()
-        resource_location = location_query.filter(ResourceModel.id == resource_id).first()  # noqa: E501
+        resource_location = location_query.filter(
```

Comment:
Resources have states too, shouldn't we only update the active resources. Same as delete?

Reply:
True @C3real-kill3r, most of the queries in the codebase were not updated on this, thanks for pointing that out. In case you see any other query like this kindly reach out.